### DISCUSSION TITLE
Status Update: 2022-11-09

### DISCUSSION THREAD

**Discussion by haixuanTao** [dora-rs/dora#139] [Status Update]
:robot:

**Comment by haixuanTao** [comment]
## dora-rover

Control the car is now ok. We have succeeded in overriding the control and are currently using simulated radio control (rc).
- Fixed latency issues with new Nvidia AGX Orin. 
- Fixed issues with Px4 firmware and calibration that were not present in AGX Xavier. ( Thanks to Bob )
- Changed the planner from hybrid_astar to [frenet](https://github.com/erdos-project/frenet_optimal_trajectory_planner). This makes the waypoints a lot more stable at the cost of a more complex planner. The stability is really useful as the localization is noisier in real life.
- Ported both `yolov5` and `yolop` on the rover. 
- Still need to do more field tests, but we can now run the rover with planned directions. 
- Ported `obstacle_location` operator from `dora-drives` that combines LIDAR + Obstacles bounding box to find the exact position of the obstacles. Porting has been more difficult as there is significantly more noise in real-life LIDAR than in simulation LIDAR. 
-> Obstacle avoidance trajectory is next step.

**Comment by phil-opp** [comment]
## dora-core

We merged the following changes to `dora` in the past two weeks:

### CLI

- https://github.com/dora-rs/dora/pull/117
- https://github.com/dora-rs/dora/pull/114
- https://github.com/dora-rs/dora/pull/116
- https://github.com/dora-rs/dora/pull/123
- https://github.com/dora-rs/dora/pull/127
- https://github.com/dora-rs/dora/pull/128

### APIs

- https://github.com/dora-rs/dora/pull/120
- We also have an open PR for C++ operators: https://github.com/dora-rs/dora/pull/121
  - There are a some link errors in the Windows example, which we decided to ignore for now, as Windows isn't our focus at the moment.

### Dataflow YAML

- https://github.com/dora-rs/dora/pull/124
- https://github.com/dora-rs/dora/pull/129

### Fixes

- https://github.com/dora-rs/dora/pull/115
  - Fixes the graph on macOS, where opening HTML files from the temp directory is apparently not permitted.
